### ISSUE TITLE
When fh is left to default in pycaret.time_series R2 is NaN in compare_models

### ISSUE THREAD

**Issue by moezali1** [pycaret/pycaret#1697]
When `fh` is left to default i.e. 1 in pycaret.time_series `setup`, R2 is NaN in the `compare_models` or other training functions such as `create_model`.

My understanding is this is happening because R2 cannot be calculated only on 1 point. 

We can discuss if we should change the default `fh` value in setup to avoid wrong impression in subsequent functions.

![image](https://user-images.githubusercontent.com/54699234/137588660-3021447e-bff5-49cf-8560-0da1ae432cae.png)
![image](https://user-images.githubusercontent.com/54699234/137588666-922e144c-5e72-4103-974a-828fcca01c19.png)

**Comment by ngupta23** [comment]
I do not think we should change the default `fh`. Instead if fh = 1, we should remove R2 from metrics. Are you ok with that?
